test/layers/sanitizer/CMakeLists.txt

```diff
@@ -34,4 +34,6 @@ function(add_sanitizer_test name)
     set_sanitizer_test_properties(${name})
 endfunction()
 
-add_sanitizer_test(asan asan.cpp)
+if(ANY_ADAPTER)
```

Comment:
not too sure but since this is linking `${PROJECT_NAME}::mock` above I suspect it might be designed to work with the mock adapter, some of the layer and tracing tests are

Reply:
Just had a look and it seems like the test is designed to run with a normal adapter, so this change does make sense I think. Probably the mock inclusion was just copy/paste. I'm not seeing a segfault with any adapter including the mock adapter, and the test is pretty trivial so might be worth looking into that - could be something dodgy with your env

Reply:
Reverted it for now, we can look at it more if we decide to expand the testing later.